northstar_tests/tests/tests.rs

```diff
@@ -0,0 +1,323 @@
+});
+
+// The test is flaky and needs to listen for notifications
+// in order to be implemented correctly
+test!(crashing_container, {
+    let mut runtime = Northstar::launch().await?;
+
+    // install test container
+    runtime.install_test_container().await?;
+    runtime.install_test_resource().await?;
+
+    for _ in 0..10 {
+        runtime.test_cmds("crash").await;
+        runtime.start(TEST_CONTAINER).await?;
+        runtime
+            .assume_notification(
+                |n| {
+                    n == &Notification::Exit {
+                        container: TEST_CONTAINER.try_into().unwrap(),
+                        status: ExitStatus::Exit(254),
+                    }
+                },
+                15,
+            )
+            .await?;
+    }
+
+    runtime.uninstall_test_container().await?;
+    runtime.uninstall_test_resource().await?;
+
+    runtime.shutdown().await
+});
+
+// Check the uid/gid of a started container
+test!(check_uid_and_gid, {
+    let runtime = Northstar::launch().await?;
+
+    runtime.install_test_container().await?;
+    runtime.install_test_resource().await?;
+
+    runtime.test_cmds("whoami").await;
+
+    runtime.start(TEST_CONTAINER).await?;
+    logger::assume("uid: 1000, gid: 1000", 5).await?;
+    runtime.stop(TEST_CONTAINER, 5).await?;
+
+    runtime.uninstall_test_container().await?;
+    runtime.uninstall_test_resource().await?;
+
+    runtime.shutdown().await
+});
+
+// Check whether after a runtime start, container start and shutdow
+// any filedescriptor is leaked
+test!(fd_leak, {
+    /// Collect a set of files in /proc/$$/fd
+    fn fds() -> Result<Vec<PathBuf>, std::io::Error> {
+        let mut links = std::fs::read_dir("/proc/self/fd")?
+            .filter_map(Result::ok)
+            .flat_map(|entry| entry.path().read_link())
+            .collect::<Vec<_>>();
+        links.sort();
+        Ok(links)
+    }
+    // Collect list of fds
+    let before = fds()?;
+
+    let runtime = Northstar::launch().await?;
+    runtime.install_test_container().await?;
+    runtime.install_test_resource().await?;
+
+    runtime.start(TEST_CONTAINER).await?;
+    logger::assume("test_container: Sleeping", 5u64).await?;
+    runtime.stop(TEST_CONTAINER, 5).await?;
+
+    let result = runtime.shutdown().await;
+
+    // Compare the list of fds before and after the RT run.
+    assert_eq!(before, fds()?);
+
+    result
+});
+
+// Open many connections to the runtime
+test!(connections, {
+    let runtime = Northstar::launch().await?;
+
+    let console = runtime.config().console.as_ref().unwrap();
+
+    let mut clients = Vec::new();
+    for _ in 0..10 {
+        let client = api::client::Client::new(&console).await?;
+        clients.push(client);
+    }
+
+    let result = runtime.shutdown().await;
+
+    for client in &clients {
+        assert!(client.containers().await.is_err());
+    }
+    clients.clear();
+
+    result
+});
+
+// test!(cgroups_memory, {
```

Comment:
is this test no longer relevant?

Reply:
Not at all!

cgroups mem doesn't trigger in the GH actions runs. Commented for now and needs to be taken care of.

Reply:
Created #309